elasticsearch.go

```diff
@@ -69,7 +86,8 @@ type Config struct {
 	EnableMetrics     bool // Enable the metrics collection.
 	EnableDebugLogger bool // Enable the debug logging.
 
-	DisableMetaHeader bool // Disable the additional "X-Elastic-Client-Meta" HTTP header.
+	DisableMetaHeader  bool // Disable the additional "X-Elastic-Client-Meta" HTTP header.
+	UseHeaderCheckOnly bool
```

Comment:
This option is pretty opaque based on its name, just throwing out `CheckOnlyElasticProductHeader` for inspiration on a potential better name?